src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -96,25 +96,29 @@
 (defn create-or-reactivate-table!
   "Create a single new table in the database, or mark it as active if it already exists."
   [database {schema :schema, table-name :name, :as table}]
-  (if-let [existing-id (t2/select-one-pk Table
-                         :db_id (u/the-id database)
-                         :schema schema
-                         :name table-name
-                         :active false)]
-    ;; if the table already exists but is marked *inactive*, mark it as *active*
-    (t2/update! Table existing-id
-      {:active true})
-    ;; otherwise create a new Table
-    (let [is-crufty? (is-crufty-table? table)]
+  (let [;; if this is a crufty table, mark initial sync as complete since we'll skip the subsequent sync steps
+        is-crufty?          (is-crufty-table? table)
+        initial-sync-status (if is-crufty? "complete" "incomplete")
+        visibility-type     (when is-crufty? :cruft)]
+    (if-let [existing-id (t2/select-one-pk Table
+                                           :db_id (u/the-id database)
+                                           :schema schema
+                                           :name table-name
+                                           :active false)]
+      ;; if the table already exists but is marked *inactive*, mark it as *active*
+      (t2/update! Table existing-id
+                  {:active              true
+                   :visibility_type     (when is-crufty? :cruft)
+                   :initial_sync_status initial-sync-status})
```

Comment:
The big difference now is that previously inactive tables now have `initial_sync_status="completed"`, as long as they're not crufty.

Reply:
> The big difference now is that previously inactive tables now have initial_sync_status="completed"

in which step did we change the initial_sync_status for inactive tables during follow-up syncs?

Reply:
Oops, I meant to say the opposite - previously inactive tables should have initial_sync_status="incomplete" not "completed" at this step. I edited the original comment.